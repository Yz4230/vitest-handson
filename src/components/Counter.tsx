import { useState } from "react";

import type { FC } from "react";

const Counter: FC = () => {
  const [count, setCount] = useState(0);

  const increment = () => setCount(count + 1);

  return (
    <button
      css={{
        padding: "1em",
        border: "none",
        borderRadius: "1em",
        cursor: "pointer",
      }}
      onClick={increment}
    >
      I was clicked {count} times
    </button>
  );
};

export default Counter;
