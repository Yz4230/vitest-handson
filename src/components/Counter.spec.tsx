import { render } from "@testing-library/react";

import Counter from "./Counter";

describe("Counter", () => {
  it("should increment the count", () => {
    const { getByText } = render(<Counter />);
    const button = getByText(/I was clicked 0 times/);
    button.click();
    expect(getByText(/I was clicked 1 times/)).toBeInTheDocument();
    button.click();
    expect(getByText(/I was clicked 2 times/)).toBeInTheDocument();
  });
});
