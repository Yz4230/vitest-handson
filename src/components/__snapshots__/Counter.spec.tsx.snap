// Vitest Snapshot v1

exports[`Counter > snapshot 1`] = `
.emotion-0 {
  padding: 1em;
  border: none;
  border-radius: 1em;
  cursor: pointer;
}

<div>
  <button
    class="emotion-0"
  >
    I was clicked 
    0
     times
  </button>
</div>
`;
